### PR TITLE
Release v3.4 release notes

### PR DESCRIPTION
**Pull request by fenryka**
With RC02 cut, changes to the usual release suspects. Keeping it fairly low key as the majority of the changes are doc fixes and the bug fixes / enhancements to Karen and Charles.

Killed some of the Pre V3 elements of hte files as they add no value at this point

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -3,6 +3,27 @@ Release notes
...
 .. _release_notes_v3_3:
 
+Release 3.4
+-----------
+
+In response to the recently released [`Corda 4.0 <https://docs.corda.net/releases/release-V4.0/release-notes.html>`_] Corda 3.4 brings a number of fixes
+that address several interoperability issues we've discovered between nodes. These should make the deployment of mixed version networks far smoother
+and the upgrade path from 3 to 4 easier.
+
+The majoritythese fixes surround the serialization and class synthesis frameworks where receiving states and messages from newer (Version 4) nodes highlighted
```
Comment (GavT) [inline docs/source/release-notes.rst]: majoritythese >> majority of these

File: docs/source/upgrade-notes.rst [hunk 2]
```diff
@@ -28,7 +28,18 @@ versions you are currently using are still in force.
 
 We also strongly recommend cross referencing with the :doc:`changelog` to confirm changes.
 
-V3.2 to v3.3
+v3.3 to v3.4
+------------
+
+* Update the Corda Release version
+
+  The ``corda_release_version`` identifier in your projects gradle file will need changing as follows:
```
Comment (GavT) [inline docs/source/upgrade-notes.rst]: projects >> project's

**Comment by fenryka** [comment]
@GavT I really just removed the pre  V3 (so 1 & 2) all the V3 notes should be there from .0 to .4

File: docs/source/release-notes.rst [hunk 3]
```diff
@@ -3,6 +3,27 @@ Release notes
...
+a few edge cases, bugs, and performant enhancements we could make.
+
+Issues Fixed
+~~~~~~~~~~~~
+
+* Don't fail deserialization with carpentry errors IFF the carpented types would be discarded through evolution [`CORDA-2704 <https://r3-cev.atlassian.net/browse/CORDA-2704>`_]
```
[the remark this answers is not in the document]
Comment (fenryka) [reply docs/source/release-notes.rst]: If was implying *if and only if*

**Comment by fenryka** [comment]
*pokes @GavT *

**Pull request merged**